tests/realtikvtest/addindextest4/partial_index_test.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2025 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package addindextest
+
+import (
+	"context"
+	"testing"
+
+	"github.com/pingcap/tidb/pkg/domain"
+	"github.com/pingcap/tidb/pkg/meta/model"
+	"github.com/pingcap/tidb/pkg/parser/ast"
+	"github.com/pingcap/tidb/pkg/testkit"
+	"github.com/pingcap/tidb/tests/realtikvtest"
+	"github.com/stretchr/testify/require"
+)
+
+func findIndex(t *testing.T, dom *domain.Domain, tableName string, indexName string) *model.IndexInfo {
+	tbl, err := dom.InfoSchema().TableByName(context.Background(), ast.NewCIStr("test"), ast.NewCIStr(tableName))
+	require.NoError(t, err)
+
+	idx := tbl.Meta().FindIndexByName(indexName)
+	return idx
+}
+
+func validatePartialIndexExists(t *testing.T, dom *domain.Domain, tableName string, indexName string) {
+	idx := findIndex(t, dom, tableName, indexName)
+	require.NotNil(t, idx)
+	require.NotEmpty(t, idx.ConditionExprString)
+}
+
+func TestPartialIndexDDL(t *testing.T) {
+	store, dom := realtikvtest.CreateMockStoreAndDomainAndSetup(t)
+	tk := testkit.NewTestKit(t, store)
+
+	t.Run("TestCreatePartialIndex", func(t *testing.T) {
```

Comment:
These tests have to be added in `realtikvtest`, because creating index is only allowed for fast-reorg. Creating index with other methods will be avoided in https://github.com/pingcap/tidb/pull/62762.